gtsam/discrete/DecisionTreeFactor.h

```diff
@@ -147,6 +148,10 @@ namespace gtsam {
     /// Calculate error for DiscreteValues `x`, is -log(probability).
     double error(const DiscreteValues& values) const override;
 
+    /// Multiply factors, DiscreteFactor::shared_ptr edition
```

Comment:
Please document behavior here, esp. regarding return type and expected complexity